Build error on CMake 4.0+

## Enhancement

Since CMake 4.0, compatibility with versions older than 3.5 has been entirely removed and will result in an error if specified. Many of third-party libraries declare cmake_minimum_required() lower than 3.5, resulting in build error:
```
[cmake] CMake Error at contrib/xxHash/cmake_unofficial/CMakeLists.txt:8 (cmake_minimum_required):
[cmake]   Compatibility with CMake < 3.5 has been removed from CMake.
[cmake] 
[cmake]   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
[cmake]   to tell CMake that the project requires at least <min> but has been updated
[cmake]   to work with policies introduced by <max> or earlier.
[cmake] 
[cmake]   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,9 @@ project (TiFlash LANGUAGES C CXX ASM)
 
 cmake_policy(SET CMP0048 NEW)
 
+# Bypass the CMake error "Compatibility with CMake < 3.5 has been removed" since CMake 4.0.
+set(CMAKE_POLICY_VERSION_MINIMUM 3.5)
+
 set(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY true)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${TiFlash_SOURCE_DIR}/cmake/Modules/")
 set(CMAKE_MACOSX_RPATH 1)

